src/docs/workflows/device-commissioning.mdx

```diff
@@ -0,0 +1,116 @@
+
+Often, users would invite technicians to come to their home or company and set up the system for them, and later on, the technician would pass the admin right to the user. Sometimes this necessitates a "handoff" experience at the end of your workflow.
+
+## Do not force them to create an account until the need comes up.
+
+Accounts are often used to control the device from anywhere (web, phone, device's HMI) or manage access for different people. However, not all of these functionalities are necessary during an initial device setup. You should only prompt your users to create the account when they actually need it, rather than making the onboarding process burdensome.
+
+## Devices can lose connection halfway through the process.
+
+You should never take for granted your devices will stay connected through the entire process. If there is a network "hiccup," how do you vent users' frustration? Would your application wait & retry in the background, pause the process, or abort the entire workflow?
+
+## It can take a long time before a device responds.
+
+The transmission between IoT devices can take two minutes for jumping between nodes. When your user taps on a button but doesn't see their thermostats working, how would you communicate this latency?
+
+## When a device is installed, its firmware might already be outdated.
+
+There is usually a time gap between when a device is boxed in the factory and when that device gets unboxed by your user. By the time the user plugs it in, your device's firmware might need a firmware upgrade.
+
+Tell your user how long the upgrade might take and whether this will impact the setup process. If the upgrade is non-interruptive, you can have the process running in the background while users are doing the configuration and return to it after they are finished with everything else.
+
+## Answer users' questions where they are asked.
+
+Your users might run into different problems while using the application, and your application should be able to help them troubleshoot before things get out of control. For example, the application expects that the internet connection indicator light is blinking, but your user doesn't see it happening, what can they do to rescue the situation? Do they need to call your company right away, restart their router, or simply move the device closer to the WiFi connection? Often, this can be tackled with a simple "Having trouble?" button on the screen that can help the user self-diagnose the issue.
+
+You should also include tooltips for technical jargon for non-technical or inexperienced users to help them with the basic concept. Your tooltips should not be longer than what a first-time user would need. You may include a link to an external documentation place if the more technical users would like to understand more technical details.
+
+If your configuration includes multiple steps running in the background, articulate the step names would help your users troubleshoot.
```

Comment:
```suggestion
Your users might encounter different problems while using your application. The application should be able to help them troubleshoot in the moment. For example, if the application says that an indicator light should be blinking but it isn't, how can you help the user proceed? Do they need to call your company right away, restart their router, or simply move the device closer to the WiFi connection? Often, this can be tackled with a simple "Having trouble?" button on the screen that can help the user self-diagnose common issues.

You should also avoid using technical jargon whenever possible and provide tooltips for any terms that users may not understand. Your tooltips should not be longer than what a first-time user would need. You may include a link to external documentation if you think technical users could benefit from more detailed information.

If your configuration includes multiple steps running in the background, articulate the step names would help your users troubleshoot.
```